### ISSUE TITLE
build is broken with protobuf

### ISSUE THREAD

**Issue by jmhodges** [bazelbuild/rules_go#1316]
I just bumped rules_go to 59327146e5395cd1773bab0107d974f660cc0852 in order to get at Go 1.9.4 and my build has started breaking with:

```
(07:15:53) INFO: Found 95 targets and 12 test targets...
(07:16:01) ERROR: /home/travis/bazel/outbase/external/com_github_golang_protobuf/proto/BUILD.bazel:3:1: GoCompile external/com_github_golang_protobuf/proto/linux_amd64_race_stripped/go_default_library~/github.com/golang/protobuf/proto.a failed (Exit 1): compile failed: error executing command 
  (cd /home/travis/bazel/outbase/execroot/founding && \
  exec env - \
  bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/linux_amd64_stripped/compile -go external/go_sdk/bin/go -root_file 'bazel-out/host/bin/external/io_bazel_rules_go/linux_amd64_stripped/stdlib~/ROOT' -goos linux -goarch amd64 '-cgo=1' -compiler_path /usr/bin -cc /usr/bin/gcc -cpp_flag -U_FORTIFY_SOURCE -cpp_flag -fstack-protector -cpp_flag -B/usr/bin -cpp_flag -B/usr/bin -cpp_flag -Wunused-but-set-parameter -cpp_flag -Wno-free-nonheap-object -cpp_flag -fno-omit-frame-pointer -cpp_flag -g0 -cpp_flag -O2 -cpp_flag '-D_FORTIFY_SOURCE=1' -cpp_flag -DNDEBUG -cpp_flag -ffunction-sections -cpp_flag -fdata-sections -cpp_flag -g0 -ld_flag -lstdc++ -ld_flag -lm -ld_flag '-fuse-ld=gold' -ld_flag -Wl,-no-as-needed -ld_flag -Wl,-z,relro,-z,now -ld_flag -B/usr/bin -ld_flag -B/usr/bin -ld_flag -pass-exit-codes -package_list external/go_sdk/packages.txt -src external/com_github_golang_protobuf/proto/clone.go -src external/com_github_golang_protobuf/proto/decode.go -src external/com_github_golang_protobuf/proto/encode.go -src external/com_github_golang_protobuf/proto/equal.go -src external/com_github_golang_protobuf/proto/extensions.go -src external/com_github_golang_protobuf/proto/lib.go -src external/com_github_golang_protobuf/proto/message_set.go -src external/com_github_golang_protobuf/proto/pointer_unsafe.go -src external/com_github_golang_protobuf/proto/properties.go -src external/com_github_golang_protobuf/proto/text.go -src external/com_github_golang_protobuf/proto/text_parser.go -o 'bazel-out/host/bin/external/com_github_golang_protobuf/proto/linux_amd64_race_stripped/go_default_library~/github.com/golang/protobuf/proto.a' -trimpath . -I . -- -p github.com/golang/protobuf/proto -race)
external/com_github_golang_protobuf/proto/clone.go:38:2: can't find import: "log"
2018/02/09 07:16:01 error running compiler: exit status 1
```

**Comment by jmhodges** [comment]
(I suspect b689873035bf9441acad187e2bb254fc403f5e2f and am running some CI jobs to confirm)

**Comment by jmhodges** [comment]
Nope, it's earlier than that. 3265fbbb95873a2e16bd8af77a5b84717be4e769, for instance, failed with the same error.

**Comment by jmhodges** [comment]
Found it! The first commit that breaks the build is 213d5fcd0853a51f970a2cdd4c984b411d5bbf79 from https://github.com/bazelbuild/rules_go/pull/1295

**Comment by jmhodges** [comment]
(It might matter that this build was with `--features=race` in it?)

**Comment by jayconrod** [comment]
Confirmed. Sorry about the break. I'm surprised we didn't have test coverage to catch this.

It appears that we're always considering the stdlib a "bootstrap" rule to avoid adding a circular dependency on itself. However, this has other implications: race mode is forced off for bootstrap rules.

I'm working on a fix.

**Comment by whitlockjc** [comment]
If you were stuck on a broken version of bazel, is there any way to work around this with some bazel flags or build config/rules?

**Comment by jayconrod** [comment]
@whitlockjc Are you on an old version of Bazel or rules_go? This was a bug in rules_go, and I wouldn't expect the Bazel version to matter too much.

Either way though, I'd encourage you to find a way to upgrade. Neither Bazel nor rules_go are supported this far back. Go 1.9 is about to stop being supported, too.

**Comment by whitlockjc** [comment]
I'm on an old version of Bazel and the project deciding to use this version of Bazel has had some difficulties upgrading.  I now see that `rules_go` isn't a Bazel thing but just another dep/tool being used.  `WORKSPACE` is pegged to `2d9f328a9723baf2d037ba9db28d9d0e30683938`, I need to figure out which version this was fixed and see if that fixes things.  Thanks for the tip.

**Comment by whitlockjc** [comment]
It looks like the first release of `rules_go` that has your fix requires Bazel `0.8.0` and I'm stuck at `0.5.4` right now.  I'll see what I can do to get the project to upgrade Bazel.
